From 36369848ff80ff910fc8b6bfd8e51671a241997a Mon Sep 17 00:00:00 2001
From: Tianshuo Deng <tdeng@twitter.com>
Date: Mon, 8 Sep 2014 14:12:11 -0700
Subject: [PATCH 46/93] update scala 2.10

Try to upgrade to scala 2.10

Author: Tianshuo Deng <tdeng@twitter.com>

Closes #35 from tsdeng/update_scala_2_10 and squashes the following commits:

1b7e55f [Tianshuo Deng] fix comment
bed9de3 [Tianshuo Deng] remove twitter artifactory
2bce643 [Tianshuo Deng] publish fix
06b374e [Tianshuo Deng] define scala.binary.version
fcf6965 [Tianshuo Deng] Merge branch 'master' into update_scala_2_10
e91d9f7 [Tianshuo Deng] update version
5d18b88 [Tianshuo Deng] version
83df898 [Tianshuo Deng] update scala 2.10

Conflicts:
	pom.xml
Resolution:
    Newline addition caused a spurrious conflict and deconflicted CDH
    version changes with Scala version update.
---
 parquet-scala/pom.xml                              |    4 ++--
 parquet-scrooge/pom.xml                            |    6 +++---
 .../parquet/scrooge/ScroogeStructConverter.java    |    3 ++-
 pom.xml                                            |    4 +++-
 4 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/parquet-scala/pom.xml b/parquet-scala/pom.xml
index 8826517..79d7703 100644
--- a/parquet-scala/pom.xml
+++ b/parquet-scala/pom.xml
@@ -8,7 +8,7 @@
 
   <modelVersion>4.0.0</modelVersion>
 
-  <artifactId>parquet-scala</artifactId>
+  <artifactId>parquet-scala_${scala.binary.version}</artifactId>
   <packaging>jar</packaging>
 
   <name>Parquet Scala</name>
@@ -42,7 +42,7 @@
     </dependency>
     <dependency>
       <groupId>org.scalatest</groupId>
-      <artifactId>scalatest_2.9.3</artifactId>
+      <artifactId>scalatest_${scala.binary.version}</artifactId>
       <version>1.9.2</version>
       <scope>test</scope>
     </dependency>
diff --git a/parquet-scrooge/pom.xml b/parquet-scrooge/pom.xml
index f537766..d1d7caf 100644
--- a/parquet-scrooge/pom.xml
+++ b/parquet-scrooge/pom.xml
@@ -8,7 +8,7 @@
 
   <modelVersion>4.0.0</modelVersion>
 
-  <artifactId>parquet-scrooge</artifactId>
+  <artifactId>parquet-scrooge_${scala.binary.version}</artifactId>
   <packaging>jar</packaging>
 
   <name>Parquet Scrooge</name>
@@ -73,8 +73,8 @@
     </dependency>
     <dependency>
       <groupId>com.twitter</groupId>
-      <artifactId>scrooge-core_2.9.2</artifactId>
-      <version>3.12.1</version>
+      <artifactId>scrooge-core_${scala.binary.version}</artifactId>
+      <version>3.16.3</version>
     </dependency>
     <dependency>
       <groupId>com.twitter</groupId>
diff --git a/parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java b/parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java
index a13c741..ebcc97e 100644
--- a/parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java
+++ b/parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java
@@ -25,6 +25,7 @@ import static parquet.thrift.struct.ThriftField.Requirement.*;
 
 import scala.collection.Iterator;
 import scala.collection.JavaConversions;
+import scala.collection.JavaConversions$;
 import scala.collection.Seq;
 import scala.reflect.Manifest;
 
@@ -75,7 +76,7 @@ public class ScroogeStructConverter {
     }
 
     List<ThriftField> children = new LinkedList<ThriftField>();//{@link ThriftType.StructType} uses foreach loop to iterate the children, yields O(n) time for linked list
-    Iterable<ThriftStructField> scroogeFields = JavaConversions.asIterable(companionObject.metaData().fields());
+    Iterable<ThriftStructField> scroogeFields = JavaConversions$.MODULE$.asJavaIterable(companionObject.metaData().fields());
     for (ThriftStructField field : scroogeFields) {
       children.add(toThriftField(field));
     }
diff --git a/pom.xml b/pom.xml
index 273e223..87329a0 100644
--- a/pom.xml
+++ b/pom.xml
@@ -121,7 +121,9 @@
     <protobuf.version>${cdh.protobuf.version}</protobuf.version>
     <avro.version>${cdh.avro.version}</avro.version>
     <scala.maven.test.skip>false</scala.maven.test.skip>
-    <scala.version>2.9.2</scala.version>
+    <scala.version>2.10.4</scala.version>
+    <!-- scala.binary.version is used for projects that fetch dependencies that are in scala -->
+    <scala.binary.version>2.10</scala.binary.version>
   </properties>
 
   <modules>
-- 
1.7.0.4

